### DISCUSSION TITLE
Does Page.add_freetext_annot() work with custom fonts?

### DISCUSSION THREAD

**Discussion by ipoyt** [pymupdf/PyMuPDF#3759] [Looking for help]
Hello!

I would love to add some text into a rectangle, but text should be automatically wrapped to a new line at box width. Lines not fitting into the box will be invisible. It's implemented via `Page.add_freetext_annot()`. 
The problem is that while using `Shape.insert_textbox()` I can use a font reference as fontname, after adding custom font files, however this is not working the same way in `Page.add_freetext_annot()`. 

This is how I add custom fonts:
```
page.insert_font(fontname=font_ref, fontfile=str(font_file.absolute()))
```

This is how it used to work properly with `Shape.insert_textbox()`:
```
        shape.insert_textbox(
            rect=rect,
            buffer=text,
            fontname=font_ref,  # the one I've added via page.insert_font
            fontsize=font_kwargs.font_size,
            lineheight=font_kwargs.line_height,
            color=font_kwargs.font_color,
            align=font_kwargs.align,
            rotate=page.rotation,
        )
```

This is what I'm trying to do with `Page.add_freetext_annot()`:
```
        self.page.add_freetext_annot(
            rect=rect,
            text=text,
            fontname=font_ref,  # the one I've added via page.insert_font
            fontsize=font_kwargs.font_size,
            # lineheight=font_kwargs.line_height,
            color=font_kwargs.font_color,
            align=font_kwargs.align,
            rotate=page.rotation,
        )
```
But unfortunately it doesn't use the file, using default fonts instead.
Is there any way to have text automatically wrapped to a new line at box width and lines not fitting into the box invisible, along with custom fonts?

Thank you!

**Comment by JorjMcKie** [comment]
This is not working yet unfortunately.
We are in the process to support arbitrary fonts and rich text in free text annotations - but we are not there yet.

**Reply by JorjMcKie** [reply]
If you don't insist on annotations, then insert_htmlbox and insert_textbox, fill_textbox are for you.
HTML box most elegantly works here and allows mixtures of fonts, colors, weights and styles at the same time. If space problems occur, automatic scaling is made.
The only difference: this content is permanent.
